### PR TITLE
collect anaconda-post log if appliance configure fails.

### PR DESCRIPTION
**Pull request by amavinag**
* while template deploy, if appliance configure fails
    collect anaconda-post.log.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 07c6e0ef7d2e5bb8bef987a384c0c0788b309e9c:

Everything seems all right :smile: .
*CFME QE Bot*

File: scripts/clone_template.py [hunk 1]
```diff
@@ -164,14 +165,32 @@ def main(**kwargs):
...
+            app = Appliance(kwargs['provider'], deploy_args['vm_name'])
+            app.configure()
+            logger.info('Successfully Configured the appliance.')
+    except Exception as e:
+        logger.exception(e)
+        logger.error('Appliance Configuration Failed')
         app = Appliance(kwargs['provider'], deploy_args['vm_name'])
-        app.configure()
+        ssh_client = app.ssh_client()
+        status, output = ssh_client.run_command('find ~ -name anaconda-post.log')
```
Comment (psav) [inline scripts/clone_template.py]: I'd replace ~ with the absolute path here.

File: scripts/clone_template.py [hunk 2]
```diff
@@ -164,14 +165,32 @@ def main(**kwargs):
...
-                              fail_condition=None)
-    logger.info('IP Address returned is {}'.format(ip))
+    try:
+        ip, time_taken = wait_for(provider.get_ip_address, [deploy_args['vm_name']], num_sec=1200,
+                                  fail_condition=None)
+        logger.info('IP Address returned is {}'.format(ip))
+    except Exception as e:
+        logger.exception(e)
+        logger.error('IP address not returned')
+        return 10
```
Comment (psav) [inline scripts/clone_template.py]: Any reason why 10 in particular?

Comment (amavinag) [inline scripts/clone_template.py]: @psav no specific reason, let me know if we need to add different return code there for better traceback?

**Comment by dajoRH** [comment]
Lint report for commit 549b894b5386fdfed213611d1cfa9941fa8bb2a4:

Everything seems all right :smile: .
*CFME QE Bot*

File: scripts/clone_template.py [hunk 3]
```diff
@@ -164,14 +165,32 @@ def main(**kwargs):
...
+            app = Appliance(kwargs['provider'], deploy_args['vm_name'])
+            app.configure()
+            logger.info('Successfully Configured the appliance.')
+    except Exception as e:
+        logger.exception(e)
+        logger.error('Appliance Configuration Failed')
         app = Appliance(kwargs['provider'], deploy_args['vm_name'])
-        app.configure()
+        ssh_client = app.ssh_client()
+        status, output = ssh_client.run_command('find /root/anaconda-post.log')
```
Comment (psav) [inline scripts/clone_template.py]: What happens if this file doesn't exist?

Comment (amavinag) [inline scripts/clone_template.py]: if not exists status == 0 fails (status returned will be 1 if does not exist)

**Pull request merged**